test/cpp/qps/client_callback.cc

```diff
@@ -148,36 +147,36 @@ class CallbackUnaryClient final : public CallbackClient {
 
  protected:
   bool ThreadFuncImpl(Thread* t, size_t thread_idx) override {
-    for (size_t i = 0; i < ctxs_[thread_idx].size(); i++) {
-      ScheduleRpc(t, thread_idx, i);
+    for (int vector_idx = thread_idx; vector_idx < total_outstanding_rpcs_;
+         vector_idx += num_threads_) {
+      ScheduleRpc(t, thread_idx, vector_idx);
     }
     return true;
   }
 
   void InitThreadFuncImpl(size_t thread_idx) override { return; }
 
  private:
-  void ScheduleRpc(Thread* t, size_t thread_idx, size_t queue_idx) override {
+  void ScheduleRpc(Thread* t, size_t thread_idx, size_t vector_idx) override {
     if (!closed_loop_) {
       gpr_timespec next_issue_time = NextIssueTime(thread_idx);
```

Comment:
You know, this NextIssueTime thing may not make sense anymore given the way that RPCs are now handled. We may need to change this, but this can hold off for now since we don't really analyze open-loop tests generally.

Reply:
And I also realize  that this is the only reason that thread_idx is being passed around everywhere. Maybe that's an incentive to fix it in this case.

Reply:
Actually, I know the right thing to do in this case. The `NextIssueTime` thing should be handled like in the sync case. The vector of NextIssueTime should be as big as the ctx_ vector and then always indexed based on the ctx_vector_idx . The initialization of the NextIssueTime vector (wherever it is, I've forgotten) should be like in the sync client, based on the total number of outstanding RPCs (which is also the thread count in the sync client, but not here).

Reply:
Yeah I need to clarify my understanding of next issue time. I believe in the case of Async where a list of next issue time is maintained, the next issue time is still generated per thread_idx. I believe this is because the Async thread issues a new RPC after the previous RPC wakes it up from waiting on the cq. This is not the case for Callback where a new RPC may be issued in a seperate Event manager thread.

I agree with Vijay though. I'll figure this out as I add more scenarios to the benchmark suite in another PR.